specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2020-04-01/examples/CosmosDBCassandraKeyspaceMigrateToManualThroughput.json

```diff
@@ -0,0 +1,26 @@
+{
+  "parameters": {
+    "accountName": "ddb1",
+    "resourceGroupName": "rg1",
+    "api-version": "2020-04-01",
+    "subscriptionId": "subid",
+    "keyspaceName": "keyspaceName"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "properties": {
+          "resource": {
+            "throughput": 400,
```

Comment:
>throughput [](start = 13, length = 10)

Some of these values are on the throughputSettings resource. We generally discourage resource modifications via POST with the exception of regenerating secrets. I think it's another reason to model this as a PUT on throughputSettings.

Reply:
Again, the response here is just to show the state of throughputSettings resource. No properties of the throughputSettings resource were written by this operation. 
Just to give context, throughput is the resource which determines the customer usage of CosmosDB, and the number consumed throughput maps to the billing, there are two ways to model throughput: manual and autoscale. The migration action is just switching between these ways. The fact that this is an action on the resource, was the reason we modeled this action as a POST.

Reply:
None of this requires a POST action, though. It could be done on the throughputSettings resource via the discriminator pattern in the swagger. (You would have an enum with Manual and Autoscale values and the schema of the rest of the payload could be selected by the enum, basically.)

---
In reply to: [438544863](https://github.com/Azure/azure-rest-api-specs/pull/9258#discussion_r438544863) [](ancestors = 438544863)

Reply:
Please see my explanation in the previous comment.